backends/pkg/reconciler/eventmesh/handler.go

```diff
@@ -32,63 +33,81 @@ func EventMeshHandler(ctx context.Context, listers Listers) func(w http.Response
 	return func(w http.ResponseWriter, req *http.Request) {
 		logger.Debugw("Handling request", "method", req.Method, "url", req.URL)
 
-		err, convertedBrokers := fetchBrokers(listers.BrokerLister, logger)
+		err, eventMesh := BuildEventMesh(listers, logger)
 		if err != nil {
-			logger.Errorw("Error fetching and converting brokers", "error", err)
+			logger.Errorw("Error building event mesh", "error", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
 
-		brokerMap := make(map[string]*Broker)
-		for _, cbr := range convertedBrokers {
-			brokerMap[cbr.GetNameAndNamespace()] = cbr
-		}
-
-		fetchedEventTypes, err := listers.EventTypeLister.List(labels.Everything())
+		err = json.NewEncoder(w).Encode(eventMesh)
 		if err != nil {
-			logger.Errorw("Error listing eventTypes", "error", err)
+			logger.Errorw("Error encoding event mesh", "error", err)
+			http.Error(w, err.Error(), http.StatusInternalServerError)
 			return
 		}
+	}
+}
 
-		logger.Debugw("Fetched event types", "event types", fetchedEventTypes)
-
-		convertedEventTypeMap := make(EventTypeMap)
-		for _, et := range fetchedEventTypes {
-			namespaceEventTypeRef := NamespaceEventTypeRef(et)
+func BuildEventMesh(listers Listers, logger *zap.SugaredLogger) (error, EventMesh) {
```

Comment:
Returning the error as first parameter seems against the normal pattern `(returnType, error)`

Reply:
This is fixed now. Mixed up stuff with Node :D